lib/p2p/packets/packetUtils.ts

```diff
@@ -1,32 +1,56 @@
 import PacketType from './PacketType';
-import Packet from './Packet';
+import Packet, { PacketInterface } from './Packet';
 import * as packetTypes from './types';
 
-const fromRaw = (type: string, packet: string): Packet | void => {
-  switch (type) {
-    case PacketType.HELLO:
-      return new packetTypes.HelloPacket(packet);
-    case PacketType.PING:
-      return new packetTypes.PingPacket(packet);
-    case PacketType.PONG:
-      return new packetTypes.PongPacket(packet);
-    case PacketType.ORDER:
-      return new packetTypes.OrderPacket(packet);
-    case PacketType.ORDER_INVALIDATION:
-      return new packetTypes.OrderInvalidationPacket(packet);
-    case PacketType.GET_ORDERS:
-      return new packetTypes.GetOrdersPacket(packet);
-    case PacketType.ORDERS:
-      return new packetTypes.OrdersPacket(packet);
-    case PacketType.GET_HOSTS:
-      return new packetTypes.GetHostsPacket(packet);
-    case PacketType.HOSTS:
-      return new packetTypes.HostsPacket(packet);
-    default:
-      return;
+class ParserError {
+  constructor(public type: ParserErrorType, public payload: string) { }
+}
+
+enum ParserErrorType {
+  UNPARSABLE_MESSAGE,
+  INVALID_MESSAGE,
+  UNKNOWN_PACKET_TYPE,
+}
+
+const fromRaw = (raw: string): Packet => {
```

Comment:
This function need to change, because now we parse the packet in order to determine the type. 
Putting the parsing code here, including the error types, isn't a good solution IMO. It should return to `Parser`.